frontend/src/auth/pages/RegisterPage.tsx

```diff
@@ -1,3 +1,32 @@
+import { z } from 'zod'
+import { useForm, SubmitHandler } from 'react-hook-form'
+import { zodResolver } from '@hookform/resolvers/zod'
+
+const SignUpSchema = z.object({
+  email: z.string().email(),
+  password: z.string().min(3).max(20),
+})
+type SignUpSchemaType = z.infer<typeof SignUpSchema>
+
 export function RegisterPage() {
-  return <div>Register</div>
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<SignUpSchemaType>({ resolver: zodResolver(SignUpSchema) })
+
+  const onSubmit: SubmitHandler<SignUpSchemaType> = (data) => console.log(data)
+
+  return (
+    <form onSubmit={handleSubmit(onSubmit)} className="form">
```

Comment:
Pure react-hook-form implementation is ok, but shadcn/ui implementation is better for maintaining
https://ui.shadcn.com/docs/components/form